src/main/scala/uk/gov/hmrc/ct/HmrcAccountsApprovalRequired.scala

```diff
@@ -0,0 +1,102 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ct
+
+import uk.gov.hmrc.ct.accounts.frs10x.retriever.{Frs10xDirectorsBoxRetriever, Frs10xFilingQuestionsBoxRetriever}
+import uk.gov.hmrc.ct.box.retriever.FilingAttributesBoxValueRetriever
+import uk.gov.hmrc.ct.box.{Calculated, CtBoolean, CtBoxIdentifier}
+import uk.gov.hmrc.ct.version.CoHoVersions.{FRS102, FRS105}
+import uk.gov.hmrc.ct.version.calculations.ReturnVersionsCalculator
+import uk.gov.hmrc.ct.version.{CoHoEquivalent, HmrcAccounts, Return}
+
+case class HmrcAccountsApprovalRequired(value: Boolean) extends CtBoxIdentifier("True if approval required for HMRC version of accounts requires") with CtBoolean
+
+object HmrcAccountsApprovalRequired extends Calculated[HmrcAccountsApprovalRequired, FilingAttributesBoxValueRetriever] with HmrcAccountsApprovalRequiredCalculator {
+
+  override def calculate(boxRetriever: FilingAttributesBoxValueRetriever): HmrcAccountsApprovalRequired = calculateApprovalRequired(boxRetriever)
+}
+
+trait HmrcAccountsApprovalRequiredCalculator extends ReturnVersionsCalculator {
+
+
+  def calculateApprovalRequired(boxRetriever: FilingAttributesBoxValueRetriever): HmrcAccountsApprovalRequired = {
+
+    val returns = doCalculation(boxRetriever)
+    val hmrcAccountsReturn = findHmrcAccountsType(returns)
+    val coHoAccountReturn = hmrcAccountsReturn.flatMap { hmrc =>
+      findMatchingCohoAccounts(returns, hmrc)
+    }
+
+    val isFrs102 = isFRS102(hmrcAccountsReturn)
+    val isFrs105 = isFRS105(hmrcAccountsReturn)
+
+    val approvalRequired = (hmrcAccountsReturn, coHoAccountReturn, isFrs102, isFrs105) match {
+      case (Some(_), None, _, _) => true
+      case (Some(hmrc), Some(coHo), true, _) => hmrcApprovalRequireForFRS102(boxRetriever, hmrc, coHo)
+      case (Some(hmrc), Some(coHo), _, true) => hmrcApprovalRequireForFRS105(boxRetriever, hmrc, coHo)
+      case _ => false
+    }
+
+    HmrcAccountsApprovalRequired(approvalRequired)
+  }
+
+  private def hmrcApprovalRequireForFRS102(boxRetriever: FilingAttributesBoxValueRetriever, hmrc: Return, coHo: Return) = {
+
+    val (drToCoHo, plToCoHo) = boxRetriever match {
+      case br: FilingAttributesBoxValueRetriever with Frs10xFilingQuestionsBoxRetriever with Frs10xDirectorsBoxRetriever => (br.ac8021().orFalse, br.acq8161().orFalse)
+      case _ => (true, true)
+    }
+
+    !drToCoHo || !plToCoHo
+  }
+
+  private def hmrcApprovalRequireForFRS105(boxRetriever: FilingAttributesBoxValueRetriever, hmrc: Return, coHo: Return) = {
+
+    val (drToHmrc, drToCoHo, plToCoHo) = boxRetriever match {
+      case br: FilingAttributesBoxValueRetriever with Frs10xFilingQuestionsBoxRetriever with Frs10xDirectorsBoxRetriever => (br.ac8023().orFalse, br.ac8021().orFalse, br.acq8161().orFalse)
+      case _ => (true, true, true)
+    }
+
+    (drToHmrc, drToCoHo, plToCoHo) match {
+      case (false, false, false) => true
```

Comment:
Not sure I can easily simplify this. There are 8 combinations and 5 of them result in false. Therefore matching on the 3 true ones seems OK.